### DISCUSSION TITLE
Async fetching items add undefined items

### DISCUSSION THREAD

**Discussion by jyrkidn** [livewire/livewire#7815] [7. Bugs (archived)] [answered]
### Livewire version

v3.4.1

### Laravel version

v10.42.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

We dispatch an event in blade and then dispatch another event in that Livewire function and have a handler from it in Blade (this is just a simplified example since we have some complex Filament fields where we saw this issue). But the `x-for` adds some undefined elements then and in browser console, you'll see `Alpine Expression Error: item is not defined`. 
Tried to reproduce this with only Alpine, but there I can't reproduce it.

![image](https://github.com/livewire/livewire/assets/2447042/93f35cdb-02b8-4041-9a35-a842fb933cbb)

### Repository or Wirebox URL

https://github.com/jyrkidn/livewire-undefined-items

### Additional information

Wirebox url: https://wirebox.app/b/xp3gd

There is a possibility it's related to https://github.com/livewire/livewire/discussions/7475, but not sure.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by gdebrauwer** [comment, marked as answer]
I created a failing test PR: https://github.com/livewire/livewire/pull/7843
